pkg/virt-api/api.go

```diff
@@ -473,6 +480,63 @@ func (app *virtAPIApp) getSelfSignedCert() error {
 	return nil
 }
 
+func (app *virtAPIApp) createWebhook() error {
+	namespace := getNamespace()
+	registerWebhook := false
+	vmPath := vmValidatePath
+
+	_, err := app.virtCli.AdmissionregistrationV1beta1().ValidatingWebhookConfigurations().Get(virtWebhookValidator, metav1.GetOptions{})
+	if err != nil {
+		if k8serrors.IsNotFound(err) {
+			registerWebhook = true
+		} else {
+			return err
+		}
+	}
+
+	// TODO update webhook if it is different than what we generate
+	if registerWebhook {
+		_, err := app.virtCli.AdmissionregistrationV1beta1().ValidatingWebhookConfigurations().Create(&admissionregistrationv1beta1.ValidatingWebhookConfiguration{
```

Comment:
Evertyime fetching it to see if it exists, doing a create if not, doing an update if yes should be good enough.